website/addons/wiki/static/WikiEditor.js

```diff
@@ -3,56 +3,150 @@
+    self.currentText = ko.observable('');
+    self.activeUsers = ko.observableArray([]);
+    self.status = ko.observable('connecting');
+
+    self.displayCollaborators = ko.computed(function() {
+       return self.activeUsers().length > 1;
+    });
+
+    self.statusDisplay = ko.computed(function() {
+        switch(self.status()) {
+            case 'connected':
+                return 'Live Editing Mode';
+            case 'connecting':
+                return 'Attempting to Connect';
+            case 'noWebSocket':
+                return 'Your browser does not support Live Editing';
+            default:
+                return 'Live Editing Unavailable';
+        }
+    });
 
-    self.changed = ko.computed(function() {
-        return self.initText !== self.wikiText();
+    self.progressBar = ko.computed(function() {
+        switch(self.status()) {
+            case 'connected':
+                return {
+                    class: "progress-bar progress-bar-success",
+                    style: "width: 100%"
+                };
+            case 'connecting':
+                return {
+                    class: "progress-bar progress-bar-warning progress-bar-striped active",
+                    style: "width: 100%"
+                };
+            default:
+                return {
+                    class: "progress-bar progress-bar-danger",
+                    style: "width: 100%"
+                };
+        }
     });
 
-    //Fetch initial wiki text
-    $.ajax({
-        type: 'GET',
-        url: url,
-        dataType: 'json',
-        cache: false,
-        success: function(response) {
-            self.initText = response.wiki_content;
-            self.wikiText(response.wiki_content);
-        },
-        error: function(xhr, textStatus, error) {
-            $osf.growl('Error','The wiki content could not be loaded.');
-            Raven.captureMessage('Could not GET get wiki contents.', {
-                url: url,
-                textStatus: textStatus,
-                error: error
-            });
+    self.modalTarget = ko.computed(function() {
+        switch(self.status()) {
+            case 'connected':
+                return '#connected-modal';
+            case 'connecting':
+                return '#connecting-modal';
+            default:
+                return '#disconnected-modal';
         }
     });
 
+    self.wikisDiffer = function(wiki1, wiki2) {
+        // Handle inconsistencies in newline notation
+        var clean1 = typeof wiki1 === 'string' ?
+            wiki1.replace(/(\r\n|\n|\r)/gm, '\n') : '';
+        var clean2 = typeof wiki2 === 'string' ?
+            wiki2.replace(/(\r\n|\n|\r)/gm, '\n') : '';
+
+        return clean1 !== clean2;
+    };
+
+    self.changed = function() {
+        return self.wikisDiffer(self.publishedText(), self.currentText());
+    };
+
+    // Fetch initial wiki text
+    self.fetchData = function(callback) {
+        $.ajax({
```

Comment:
Also, by using the Promises interface and returning the jQuery xhr, you could refactor `revertChanges` like so:

```js
self.fetchData = function() {
    return $.ajax{...};
}
self.revertChanges = function() {
    return self.fetchData()
        .done(function() {
            self.currentText(self.publishedText());
        }
}